docs/guides/key-manager/upgrade-lsp6.md

```diff
@@ -0,0 +1,244 @@
+If you don't have a Universal Profile yet, follow our previous guide [**Create a Universal Profile**](../universal-profile/create-profile.md) or look at the [_lsp-factory.js_](../../tools/lsp-factoryjs/deployment/universal-profile.md) docs.
+
+:::
+
+In this guide, we will learn how upgrade the LSP6 Key Manager of your Universal Profile to the latest version available.
+
+By the end of this guide, you will know:
+
+- How to deploy a new LSP6 Key Manager with the last updates.
+- How to transfer ownership of your Universal Profile through the old LSP6 Key Manager.
+- How to accept ownership of your Universal Profile through the new LSP6 Key Manager
+
+## Setup
+
+```shell
+npm install @lukso/lsp-smart-contracts, web3
+```
+
+## Step 1 - Create `constants.js` file and save the needed constants
+
+Create a `constants.js` file with the following constants
+
+- PRIVATE_KEY = private key of a conroller address
+- keyManagerAddress = address of the current LSP6 Key Manager
+- universalProfileAddress = address of your Universal Profile
+  This controller must have CHANGEOWNER permission
+
+After this first step you can save & close `constants.js`.
+
+```js
+export const PRIVATE_KEY = '0x...';
+export const universalProfileAddress = '0x...';
+export const keyManagerAddress = '0x...';
+```
+
+## Step 2 - Create `upgrade-lsp6.js` and get the imports
+
+Create a file with the name `upgarde-lsp6.js` and you should have the following imports on the top of the file.
+
+```js
+import LSP6KeyManager from '@lukso/lsp-smart-contracts/artifacts/LSP6KeyManager.json' assert { type: 'json' };
+import {
+  PRIVATE_KEY,
+  keyManagerAddress,
+  universalProfileAddress,
+} from './constants.js';
+import Web3 from 'web3';
+const web3 = new Web3('https://rpc.l14.lukso.network');
+```
+
+## Step 3 - Initialize the controller account
+
+In order to use the controller account you need to initialize it using web3.js.
+
+```js
+const account = web3.eth.accounts.privateKeyToAccount(PRIVATE_KEY);
+web3.eth.accounts.wallet.add(account);
+```
+
+## Step 4 - Initialize the old LSP6 Key Manager
+
+In order to transfer ownership of your Universal Profile you need to initialize your current LSP6 Key Manager.
+
+```js
+const oldKeyManager = new web3.eth.Contract(
+  LSP6KeyManager.abi,
+  keyManagerAddress,
+);
+```
+
+## Step 5 - Deploy the new LSP6 Key Manager
+
+Deploy a new LSP6 Key Manager with the latest updates.
+
+```js
+const NewKeyManager = new web3.eth.Contract(LSP6KeyManager.abi);
+const newKeyManager = await NewKeyManager.deploy({
+  data: LSP6KeyManager.bytecode,
+  arguments: [universalProfileAddress],
+}).send({
+  from: account.address,
+  gas: 3000000,
+  gasPrice: '1000000000',
+});
+
+console.log(newKeyManager._address);
+```
+
+## Step 6 - Transfer ownership of the Universal Profile
+
+Create a payload for `transferOwnership(address)` and tarnsfer the ownership of your Universal Profile from your current LSP6 Key Manager.
```

Comment:
Maybe just a minor change, but it was kinda hard to read to have "transfer ownership" right after each other
```suggestion
Create a payload for the `transferOwnership(address)` function and shift the ownership of your Universal Profile from your current LSP6 Key Manager.
```